src/backend/joanie/lms_handler/tests/test_backend_openedx.py

```diff
@@ -0,0 +1,176 @@
+        )
+        expected_json_response = {"is_active": True}
+
+        responses.add(
+            responses.GET,
+            url,
+            status=200,
+            json=expected_json_response,
+        )
+
+        backend = LMSHandler.select_lms(resource_link)
+        self.assertIsInstance(backend, OpenEdXLMSBackend)
+
+        enrollment = backend.get_enrollment(username, resource_link)
+        self.assertEqual(len(responses.calls), 1)
+        self.assertEqual(responses.calls[0].request.url, url)
+        self.assertEqual(
+            responses.calls[0].request.headers["X-Edx-Api-Key"], "a_secure_api_token"
+        )
+        self.assertIsNone(responses.calls[0].request.body)
+        self.assertEqual(enrollment, expected_json_response)
+
+    @responses.activate
+    def test_backend_openedx_get_enrollment_failed(self):
+        """
+        If retrieving course run's enrollment failed, it should return None.
+        """
+        username = "joanie"
+        resource_link = (
+            "http://openedx.test/courses/course-v1:edx+DemoX+Demo_Course/course"
+        )
+        url = (
+            "http://openedx.test/api/enrollment/v1/enrollment/"
+            "{username:s},course-v1:edx+DemoX+Demo_Course".format(username=username)
+        )
+
+        responses.add(
+            responses.GET, url, status=500, json={"error": "Something went wrong..."}
+        )
+
+        backend = LMSHandler.select_lms(resource_link)
+        self.assertIsInstance(backend, OpenEdXLMSBackend)
+
+        enrollment = backend.get_enrollment(username, resource_link)
+        self.assertEqual(len(responses.calls), 1)
+        self.assertEqual(responses.calls[0].request.url, url)
+        self.assertEqual(
+            responses.calls[0].request.headers["X-Edx-Api-Key"], "a_secure_api_token"
+        )
+        self.assertIsNone(responses.calls[0].request.body)
+        self.assertIsNone(enrollment)
+
+    @responses.activate
+    def test_backend_openedx_set_enrollment_successfully(self):
+        """
+        Updating a user's enrollment to a course run should return a boolean
+        corresponding to the new enrollment active status.
+        True: user has been enrolled
+        False: user has been unrolled
+        """
+        username = "joanie"
+        resource_link = (
+            "http://openedx.test/courses/course-v1:edx+DemoX+Demo_Course/course"
+        )
+        url = "http://openedx.test/api/enrollment/v1/enrollment"
+        enrollment_state = random.choice([True, False])
+        expected_json_response = {"is_active": enrollment_state}
+
+        responses.add(
+            responses.POST,
+            url,
+            status=200,
+            json=expected_json_response,
+        )
+
+        backend = LMSHandler.select_lms(resource_link)
+        self.assertIsInstance(backend, OpenEdXLMSBackend)
+
+        course_id = backend.extract_course_id(resource_link)
```

Comment:
IMO the `course_id` should not be retrieved. You should line 132 directly set the expected `course_id`. If here the `extract_course_id` returns something you don't expect you will not catch it.

Reply:
Ok. You reminded me that I should test `extract_course_id`.